crates/consensus/src/transaction/legacy.rs

```diff
@@ -510,19 +510,33 @@ pub mod signed_legacy_serde {
         D: serde::Deserializer<'de>,
     {
         let SignedLegacy { tx, signature, hash } = SignedLegacy::deserialize(deserializer)?;
-        let (parity, chain_id) = from_eip155_value(signature.v.to())
-            .ok_or_else(|| serde::de::Error::custom("invalid EIP-155 signature parity value"))?;
-
-        // Note: some implementations always set the chain id in the response, so we only check if
-        // they differ if both are set.
-        if let Some((tx_chain_id, chain_id)) = tx.chain_id().zip(chain_id) {
-            if tx_chain_id != chain_id {
-                return Err(serde::de::Error::custom("chain id mismatch"));
+        let is_fake_signature =
+            signature.r.is_zero() && signature.s.is_zero() && signature.v.is_zero();
```

Comment:
could we also reference the reported bug in a comment here and document that this happens for pre bedrock transactions for example